### PR TITLE
PWX-4147 Add support for openshift

### PR DESCRIPTION
**Pull request by harsh-px**

### CODE REVIEW

File: px-spec-websvc/px-spec-websvc.go [hunk 1]
```diff
@@ -29,6 +30,9 @@ var (
 	PXTAG string
 )
 
+// kbVerRegex matches "1.7.9+coreos.0", "1.7.6+a08f5eeb62", "v1.7.6+a08f5eeb62", "1.7.6"
+var kbVerRegex = regexp.MustCompile(`.*(\d+\.\d+\.\d+).*`)
+
```
Comment (zoxpx) [inline px-spec-websvc/px-spec-websvc.go]: Please use a more precise regexp.  I recommend the following:

```go
// note, only v and spaces are a legal prefix, which will be ignored
// we are _optionally_ capturing the suffix
var kbVerRegex = regexp.MustCompile(`^[v\s]*(\d+\.\d+\.\d+)(.*)*`)
```
Tests:
> kbVerRegex.FindStringSubmatch("v1.6.11-gke.0") == []string{"v1.6.11-gke.0", "1.6.11", "-gke.0"}
> kbVerRegex.FindStringSubmatch("1.7.98") == []string{"1.7.98", "1.7.98", ""}
> kbVerRegex.FindStringSubmatch("  v1.7.6+a08f5eeb62") == []string{"  v1.7.6+a08f5eeb62", "1.7.6", "+a08f5eeb62"}

.. we will need this, to automatically recognize the "-gke*" versions.

File: px-spec-websvc/px-spec-websvc.go [hunk 2]
```diff
@@ -115,14 +119,18 @@ func generate(templateFile string, p *Params) (string, error) {
...
-	// * [1.6 docs] https://v1-6.docs.kubernetes.io/docs/admin/authorization/rbac "As of 1.6 RBAC mode is in beta"
 	p.KubeVer = strings.TrimSpace(p.KubeVer)
-	if len(p.KubeVer) > 1 && p.KubeVer[0] == 'v' { // cut out 'v' if left accidentally
-		p.KubeVer = p.KubeVer[1:]
+	if len(p.KubeVer) > 1 { // parse the actual k8s version stripping out unnecessary parts
+		matches := kbVerRegex.FindStringSubmatch(p.KubeVer)
+		if len(matches) > 1 {
+			p.KubeVer = matches[1]
+		}
 	}
```
Comment (zoxpx) [inline px-spec-websvc/px-spec-websvc.go]: I think we should fail if we have a valid KubeVer variable, but can't parse it.

Note, we already have the checks for invalid fields, ie:

`curl -v 'http://install.portworx.com/?foo=bar'`
> < HTTP/1.1 400 Bad Request
> Could not decode form: schema: invalid path "foo"

File: px-spec-websvc/templates/k8s-master-worker-response.gtpl [hunk 3]
```diff
@@ -1,4 +1,85 @@
...
+          timeoutSeconds: 15
+        name: portworx-pvc-controller-manager
+        resources:
+          requests:
+            cpu: 200m
+      hostNetwork: true
+      serviceAccountName: portworx-pvc-controller-account
+---
+{{end}}
 apiVersion: v1
```
Comment (zoxpx) [inline px-spec-websvc/templates/k8s-master-worker-response.gtpl]: just a quick note (nothing to fix yet) -- the top part will need to be shared w/ GKE-version

File: px-spec-websvc/px-spec-websvc.go (hunk 2 above)
Comment (harsh-px) [reply px-spec-websvc/px-spec-websvc.go]: Added error message in incremental

File: px-spec-websvc/px-spec-websvc.go (hunk 1 above)
Comment (harsh-px) [reply px-spec-websvc/px-spec-websvc.go]: Good idea. Updated in incremental.

File: px-spec-websvc/templates/k8s-master-worker-response.gtpl (hunk 3 above)
Comment (harsh-px) [reply px-spec-websvc/templates/k8s-master-worker-response.gtpl]: Yes, so the if condition will be Openshift || GKE.

**Pull request merged**